undertow/src/main/java/org/wildfly/extension/undertow/security/jaspi/JASPIAuthenticationMechanism.java

```diff
@@ -135,7 +135,7 @@ public AuthenticationMechanismOutcome authenticate(final HttpServerExchange exch
                 cache = Boolean.valueOf((String)registerObj);
             }
             sc.authenticationComplete(authenticatedAccount, authType, cache);
-        } else if (isValid && authenticatedAccount == null && !isMandatory(requestContext)) {
+        } else if (!isMandatory(requestContext)) {
             outcome = AuthenticationMechanismOutcome.NOT_ATTEMPTED;
```

Comment:
I've just taken a look at the spec and it says an AuthModule may return FAILURE or even throw an AuthException for non mandatory resources and if it does that then the runtime must NOT proceed with the service invocation. Here's a quote from the spec: "If the module
throws an AuthException, or returns any value other that AuthStatus.SUCCESS, the runtime must NOT
proceed to the service invocation. The runtime must process an AuthException as described above for a
request with a mandatory requestPolicy. The runtime must process any return value other than
AuthStatus.SUCCESS as it would be processed if it were returned for a request with a mandatory
requestPolicy."
So if I'm reading this correctly the else clause above must make sure that isValid is true before returning NOT_ATTEMPTED, shouldn't it?